test/rules/no-boolean-compare/test.ts.lint

```diff
@@ -0,0 +1,30 @@
+declare const x: boolean;
```

Comment:
add cases for more complex expressions like functions, result of binary boolean expression, and negating a boolean